pkg/test/resource_creator.go

```diff
@@ -31,7 +31,7 @@ func (ctx *TestCtx) GetNamespace() (string, error) {
 	if ctx.Namespace != "" {
 		return ctx.Namespace, nil
 	}
-	if Global.InCluster {
+	if *Global.SingleNamespace {
 		ctx.Namespace = os.Getenv(TestNamespaceEnv)
```

Comment:
Check if env is set and is not empty otherwise return error. We always expect `len(os.Getenv(TestNamespaceEnv)) >0` for the single namespace case.